### PR TITLE
Improved detection of unsupported iOS devices

### PR DESCRIPTION
**Pull request by cbracken**
* Detects iPad 2 and iPad Retina as unsupported devices.
* Simplifies blacklisting logic.
* Minor improvements to error messages.

### CODE REVIEW

**Comment by cbracken** [comment]
Fixes #4621 (which was originally fixed by #4783, but later re-broken)

File: packages/flutter_tools/lib/src/ios/simulators.dart [hunk 1]
```diff
@@ -380,23 +379,23 @@ class IOSSimulator extends Device {
...
 
-    // The 's' subtype of 5 is compatible.
-    if (name.contains('iPhone 5s'))
-      return true;
+    // Check for unsupported iPhones.
+    Match iPhoneMatch = new RegExp(r'iPhone [0-5]').firstMatch(name);
+    if (iPhoneMatch != null) {
+      if (name == 'iPhone 5s')
+        return true;
+      _supportMessage = 'Flutter does not support iPhones earlier than iPhone 5. Select an iPhone 5s or above.';
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: Maybe does not support iPhone 5 or earlier?

**Review by chinmaygarde — APPROVED** [review]
lgtm

**Comment by eseidelGoogle** [comment]
I would soften the language a little.  It would be trivial for us to support some of these older ipad/iphone devices.  https://github.com/flutter/flutter/issues/2089 is the bug.  I might consider referencing the bug and/or adding words like "yet" to the part which talks about the 32-bit ARM devices.

**Comment by cbracken** [comment]
@eseidelGoogle added some 'yet's to the older iPad/iPhone verbiage.

**Comment by cbracken** [comment]
Added unit tests.

File: packages/flutter_tools/lib/src/ios/simulators.dart [hunk 2]
```diff
@@ -380,23 +379,23 @@ class IOSSimulator extends Device {
...
 
-    // The 's' subtype of 5 is compatible.
-    if (name.contains('iPhone 5s'))
-      return true;
+    // Check for unsupported iPhones.
+    Match iPhoneMatch = new RegExp(r'iPhone [0-5]').firstMatch(name);
+    if (iPhoneMatch != null) {
+      if (name == 'iPhone 5s')
+        return true;
+      _supportMessage = 'Flutter does not support yet iPhone 5 or earlier. Select an iPhone 5s or above.';
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: yet support, not support yet :-)

Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/simulators.dart]: That's what I get for being overzealous with vim 'redo command'. Fixed in #7859 along with the Mac host msg.

**Pull request merged**